pkg/bindings/images/build.go

```diff
@@ -637,6 +637,20 @@ func prepareSecrets(secrets []string, contextDir string, tempManager *remote_bui
 					// add tmp file to context dir
 					tarContent = append(tarContent, tmpSecretFilePath)
 
+					modifiedSrc := fmt.Sprintf("src=%s", filepath.Base(tmpSecretFilePath))
+					modifiedOpt = append(modifiedOpt, modifiedSrc)
+				}  else if opt == "env" {
+					// read specified env into a tmp file
+					// move tmp file to tar and change secret source to relative tmp file
+					secretVal := os.Getenv(val)
+					tmpSecretFilePath, err := tempManager.CreateTempFileFromReader(contextDir, "podman-build-secret-*", strings.NewReader(secretVal))
```

Comment:
Looks like this could leak - a `defer os.Remove(...)` after the error check would fix that